full-service/src/service/decorated_types.rs

```diff
@@ -284,3 +286,92 @@ pub struct JsonProof {
     pub txo_id: String,
     pub proof: String,
 }
+
+#[derive(Deserialize, Serialize, Default, Debug)]
+pub struct StringifiedJsonUnspentTxOut {
```

Comment:
This doesn't feel good to me. I generally prefer things with real types over String-y data, so "StringifiedJson..." feels like a double red flag.

I may be missing the point, but I think I'd rather take the approach that these "decorated types" are only used to translate json-rpc API requests and responses to/from JSON. In that case, it seems like these types can have more strongly typed fields. It might also be worth using something like https://github.com/Keats/validator to enforce constraints on user-supplied data.

Reply:
Tangentially, I started moving the decorated_types into separate files under a new json_rpc module in the mock wallet PR, for example:

https://github.com/mobilecoinofficial/full-service/pull/24/files?file-filters%5B%5D=.rs&hide-deleted-files=true#diff-7456eb03019837a851279b592b4477613c877fb2592dd07d364bf8b831b6bbbd

As future work, I was sort of imagining replacing the String-y fields with stricter fields, adding more tests and fuzzing around coversion to/from JSON, etc.

Reply:
Let's chat about this - the issue is that json and javascript don't support u64, so if you naively try to return u64s, you could lose precision. The rust types are the strongly typed structs.

Reply:
Hmm. So, I think its fine to represent a 64-bit number as a String in JSON, but I don't think we need (or want) to use String on the Rust side for numbers -- even in these objects that basically exist to define the JSON schema of the API. It seems like Serde has the flexibility to serialize/deserialize a u64 field to/from a String in JSON by using field attributes, https://serde.rs/field-attrs.html. It looks like people have already fought this battle, and there is code for doing what we want:

https://github.com/serde-rs/json/issues/412
https://github.com/vityafx/serde-aux/blob/master/src/field_attributes.rs#L108

It looks like there are a number of places (pardon the pun) in this file where integers are represented as Strings. I'm sure we could manually back and forth between integers and strings (as well as documenting what each String is supposed to encode, and unit testing the conversions, etc), but that sounds tedious and error-prone compared to using Serde's field attributes to declare that a u64 should be converted to/from a JSON String.

Reply:
That's neat! We should probably use those field attrs in mobilecoind-json as well.

I created a ticket for this ([FS-90](https://mobilecoin.atlassian.net/browse/FS-90), as I think this big of a change is outside the scope of this PR (for consistency, all types should be made to use the field attrs, as well as the mobilecoind-json types we are importing).

I also created a ticket for re-implementing the json types from mobilecoind-json, rather than importing them. However, I do think we should discuss, as it is nice to have a consistent object API across our wallet implementations. [FS-91](https://mobilecoin.atlassian.net/browse/FS-91).

For the purposes of this PR (to unblock the electron wallet integration), I think we should move forward with this bandaid, and we can do full surgery later, possibly as part of the PR that is refactoring the json types. The refactor should not have any affect on downstream.